### PR TITLE
Decompile mystery event script

### PR DESCRIPTION
**Pull request by DizzyEggg**

### CODE REVIEW

File: asm/macros/event.inc [hunk 1]
```diff
@@ -262,6 +262,13 @@
 	.2byte \output
 	.2byte SPECIAL_\function
 	.endm
+	
+	@ temporary solution
+	.macro specialvar_ output, functionId
```
Comment (Diegoisawesome) [inline asm/macros/event.inc]: What's this for?

Comment (DizzyEggg) [reply asm/macros/event.inc]: oh, it was something I tried to do before the data was dumped and forgot to remove it

File: src/mystery_event_script.c [hunk 2]
```diff
@@ -0,0 +1,396 @@
...
+    u8 value = ScriptReadByte(ctx);
+    ctx->data[2] = value;
+    return FALSE;
+}
+
+bool8 MEScrCmd_setmsg(struct ScriptContext *ctx)
+{
+    u8 value = ScriptReadByte(ctx);
+    u8 *str = (u8 *)(ScriptReadWord(ctx) - ctx->data[1] + ctx->data[0]);
+    if (value == 255 || value == ctx->data[2])
```
Comment (Diegoisawesome) [inline src/mystery_event_script.c]: 255->-1

File: src/mystery_event_script.c [hunk 3]
```diff
@@ -0,0 +1,396 @@
...
+    *a0 = sMysteryEventScriptContext.data[2];
+
+    return ret;
+}
+
+u32 RunMysteryEventScript(u8 *script)
+{
+    struct ScriptContext *ctx = &sMysteryEventScriptContext;
+    InitMysteryEventScript(ctx, script);
+    while (RunMysteryEventScriptCommand(ctx))
```
Comment (Diegoisawesome) [inline src/mystery_event_script.c]: Convention so far has been to keep the semicolon on the same line as the while statement, but visually separate the while statement from the surrounding lines.

**Pull request merged**